src/lib/qboard.ts

```diff
@@ -57,6 +59,8 @@ export default class QBoard {
     public canvasWidth: number,
     public canvasHeight: number
   ) {
+    this.document = baseCanvasElement.ownerDocument;
```

Comment:
is it necessary to pass this down? can't we access document without having to pass it

Reply:
You can, but I think it's bad practice. This is a slightly lesser concern with `document` than `window`, since function declarations don't automatically get directly added to `document` (it does still change since `window` is at least partially accessible through `document`, but it doesn't add a new direct property).

The main issue is that some external method might in the future treat something that accesses `document` as a public method, but then they don't know that it has a dependency on `document`. This is like having an API which internally relies on client system time when doing so is unnecessary. It also makes the block of code not unit-testable.

Reply:
again, i refuse to complicate code for this reason